src/blockchains/cosmos.ts

```diff
@@ -0,0 +1,82 @@
+import { BlockchainHandler, BlockchainHandlerOpts } from '../blockchain-handler';
+import { AccountID } from 'caip';
+import { getConsentMessage, LinkProof } from '../utils';
+import { signTx, Tx, SignMeta, verifyTx } from '@tendermint/sig';
+const { base64ToBytes, bytesToBase64, toCanonicalJSONBytes } = require('@tendermint/belt');
+import * as uint8arrays from 'uint8arrays';
+import { sha256  } from 'js-sha256';
+
+const namespace = 'cosmos';
+
+const stringEncode = (str: string): string => bytesToBase64(uint8arrays.fromString(str));
+
+// return data in the cosmos unsigned transaction format
+function asTransaction(address: string, message: string): Tx {
+  return {
+    fee: {
+      amount: [{ amount: '0', denom: '' }],
+      gas: '0',
+    },
+    memo: message,
+    msg: [
+      {
+        type: 'cosmos-sdk/MsgSend',
+        value: {
+          from_address: address,
+          to_address: address,
+          amount: [{ amount: '0', denom: '0' }],
+        },
+      },
+    ],
+  };
+}
+
+// generate metadata for signing the transaction
+function getMetaData(): SignMeta {
+  return {
+    account_number: '1',
+    chain_id: 'cosmos',
+    sequence: '0',
+  };
+}
+
+async function createLink(did: string, account: AccountID, provider: any, opts: BlockchainHandlerOpts): Promise<LinkProof> {
+  const { message, timestamp } = getConsentMessage(did, !opts?.skipTimestamp);
+  const encodedMsg = stringEncode(message);
+  const res = await signTx(asTransaction(account.address, encodedMsg), getMetaData(), provider);
```

Comment:
Doesn't seem like this would allow you to sign the tx remotely. The `signTx` function seems to require a wallet instance. For example, this would not work though wallet connect I assume?

Reply:
Yes that's correct. I'm not sure any remote signer implementations exist for cosmos.

Reply:
Ok, I would prefer if this was refactored such that any signature logic happens outside of the createLink function with a remote call. Once there is a standard for this type of thing within cosmos, we'll have to update it here to conform to that standard. See also this PR: https://github.com/ceramicnetwork/js-3id-blockchain-utils/pull/46 where we separate out the createLink logic from the verification logic (create should be light on dependencies, validate is not).

Happy to merge this PR once this is fixed here 👍

Reply:
Sure no problem, I'll make the updates.